pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex_multicluster_test.go

```diff
@@ -340,6 +340,30 @@ func TestMulticlusterAmbientIndex_ServicesForWaypoint(t *testing.T) {
 		Addresses: []string{"10.0.0.1"},
 		Network:   testNW,
 	}
+	t.Run("waypoint eds", func(t *testing.T) {
+		s := newAmbientTestServer(t, testC, testNW, "")
+		s.addService(t, "svc1",
+			map[string]string{
+				label.IoIstioUseWaypoint.Name: "wp",
+				"istio.io/global":             "true",
+			},
+			map[string]string{},
+			[]int32{80}, map[string]string{"app": "app1"}, "11.0.0.1")
+		s.assertEvent(s.t, s.svcXdsName("svc1"))
+
+		s.addWaypointSpecificAddress(t, "", s.hostnameForService("wp"), "wp", constants.AllTraffic, true)
+		s.addService(t, "wp",
+			map[string]string{},
+			map[string]string{},
+			[]int32{80}, map[string]string{"app": "waypoint"}, "10.0.0.1")
+		s.assertEvent(s.t, s.svcXdsName("svc1"))
+
+		s.addPodsForClient(t, "10.0.1.5", "wp-pod", "sa1",
```

Comment:
In addition to checking for an event in response to a new pod is it meaningful to also check for an additional update?